qualtran/bloqs/swap_network.py

```diff
@@ -116,20 +138,44 @@ def bloq_counts(
 
 
 @frozen
-class SwapWithZero(Bloq):
+class SwapWithZero(GateWithRegisters):
```

Comment:
What's going on here? Am I wrong that a GateWithRegisters should implement a decompose_from_registers method and not a build_composite bloq method?

Reply:
It can implement any, and both will work for both the use cases. You'll get the correct expected behavior for both `cirq.Circuit(cirq.decmopose_once(op))` and `show_bloq(bloq.decompose_bloq())`, i.e. it doesn't matter whether the bloq author implements `build_composite_bloq` or `decompose_from_registers`, a Bloq user would be able to do

1. For `cirq.Circuit(cirq.decmopose_once(op))`: And get a decomposed circuit with cirq style native operations, except only for those bloqs which are not defined as `GateWithRegisters`. These bloqs will get wrapped in a `BloqAsCirqGate(bloq)`
2. For `bloq.decompose_bloq()`: You'll get bloq native objects, except for cirq gates which are not defined as `GateWithRegisters`. These gates will get wrapped in a `CirqGateAsBloq(gate)` wrapper. 


So, from our perspective, it is always strictly better to define every new bloq as a `GateWithRegisters` and we can override any of `build_composite_bloq` or `decompose_from_registers` as per our choice and the other side of the interop we get for free.